Content.Shared/Pinpointer/SharedPinpointerSystem.cs

```diff
@@ -27,33 +28,75 @@ private void OnAfterInteract(EntityUid uid, PinpointerComponent component, After
         args.Handled = true;
-        component.Target = args.Target;
-        _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(args.User):player} set target of {ToPrettyString(uid):pinpointer} to {ToPrettyString(component.Target.Value):target}");
+
+        if (component.CanRetarget && !component.IsActive || _tagSystem.HasTag(args.Target.Value, "PinpointerScannable"))
+        {
+            component.Target = args.Target;
+        }
+        else
+        {
+            var pinpointerScanEvent = new GotPinpointerScannedEvent(uid);
+            RaiseLocalEvent(args.Target.Value, ref pinpointerScanEvent);
+
+            if (component.Target == null || !pinpointerScanEvent.Handled)
+                return;
+        }
+
+        if(component.StoredTargets.Contains(component.Target.Value))
+            return;
+
+        if (component.StoredTargets.Count >= component.MaxTargets)
+        {
+            _popup.PopupClient(Loc.GetString("target-pinpointer-full"),args.User,args.User);
+            return;
+        }
+
+        StoreTarget(component.Target.Value, uid, component, args.User);
+
+        _popup.PopupClient(Loc.GetString("target-pinpointer-stored", ("target", component.Target.Value)), args.User, args.User);
+
         if (component.UpdateTargetName)
             component.TargetName = component.Target == null ? null : Identity.Name(component.Target.Value, EntityManager);
     }
 
     /// <summary>
     ///     Set pinpointers target to track
     /// </summary>
-    public virtual void SetTarget(EntityUid uid, EntityUid? target, PinpointerComponent? pinpointer = null)
+    public virtual void SetTarget(EntityUid uid, EntityUid? target, PinpointerComponent? pinpointer = null, EntityUid? user = null)
     {
         if (!Resolve(uid, ref pinpointer))
             return;
 
-        if (pinpointer.Target == target)
+        if (pinpointer.Target == target || target == null)
             return;
 
         pinpointer.Target = target;
-        if (pinpointer.UpdateTargetName)
-            pinpointer.TargetName = target == null ? null : Identity.Name(target.Value, EntityManager);
-        if (pinpointer.IsActive)
-            UpdateDirectionToTarget(uid, pinpointer);
+
+        //Searches for the name of the tracked entity
+        if (pinpointer.Target != null)
+        {
+            pinpointer.TargetName = Identity.Name(pinpointer.Target.Value, EntityManager);
+        }
+
+        if (user != null && pinpointer.Target != null)
+        {
+            if (pinpointer.TargetName != null)
+            {
+                _popup.PopupEntity(Loc.GetString("targeting-pinpointer-succeeded",
+                    ("target", pinpointer.TargetName)), user.Value, user.Value);
+            }
+
+        }
+
+        //Turns on the pinpointer if the target is changed through the verb menu.
+        if (!pinpointer.IsActive && user != null)
+        {
+            TogglePinpointer(uid, pinpointer);
+        }
+
```

Comment:
user it not even being used here so no point checking if it's not null.

Reply:
It's a leftover from old code(it searching automatically on FTL travel) that I forgot to remove